remote-api/src/main/java/org/alfresco/rest/api/impl/RulesImpl.java

```diff
@@ -79,23 +80,25 @@ public Rule getRuleById(final String folderNodeId, final String ruleSetId, final
         final NodeRef ruleSetNodeRef = validateRuleSetNode(ruleSetId, folderNodeRef);
         final NodeRef ruleNodeRef = validateRuleNode(ruleId, ruleSetNodeRef);
 
-        return Rule.from(ruleService.getRule(ruleNodeRef));
+        return Rule.from(ruleService.getRule(ruleNodeRef), isRuleSetNotNullAndShared(ruleSetNodeRef));
     }
 
     @Override
     public List<Rule> createRules(final String folderNodeId, final String ruleSetId, final List<Rule> rules)
     {
         final NodeRef folderNodeRef = validateFolderNode(folderNodeId);
         // Don't validate the ruleset node if -default- is passed since we may need to create it.
+        NodeRef ruleSetNodeRef = null;
         if (RuleSet.isNotDefaultId(ruleSetId))
         {
-            validateRuleSetNode(ruleSetId, folderNodeRef);
+            ruleSetNodeRef = validateRuleSetNode(ruleSetId, folderNodeRef);
         }
 
+        final boolean isShared = isRuleSetNotNullAndShared(ruleSetNodeRef);
```

Comment:
Will this work if `-default-` is passed in?  I think we'll end up always saying that the `-default-` rule set is not shared.

Reply:
Fixed it, assuming that missing rule set node will be created within `ruleService.saveRule(folderNodeRef, rule)`, but I'm not sure about that. Further fix might be required in future.